service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java

```diff
@@ -95,4 +95,40 @@ public void changeEmail(JWTData userData, ChangeEmailRequest changeEmailRequest)
         AuthDatabaseOperations.getFullName(user.into(Users.class)),
         changeEmailRequest.getNewEmail());
   }
+
+  @Override
+  public void changePrivilegeLevel(
+      JWTData userData, ChangePrivilegeLevelRequest changePrivilegeLevelRequest) {
+    // check if user is admin
+    if (!(userData.getPrivilegeLevel() == PrivilegeLevel.ADMIN
+        || userData.getPrivilegeLevel() == PrivilegeLevel.SUPER_ADMIN)) {
+      throw new AuthException("User does not have the required privilege level");
+    }
+
+    UsersRecord user =
+        db.selectFrom(USERS)
+            .where(USERS.EMAIL.eq(changePrivilegeLevelRequest.getTargetUserEmail()))
+            .fetchOne();
+    if (user == null) {
+      throw new UserDoesNotExistException(changePrivilegeLevelRequest.getTargetUserEmail());
+    }
+
+    // normal admins can't create super admins
+    if (userData.getPrivilegeLevel() == PrivilegeLevel.ADMIN
+        && changePrivilegeLevelRequest.getNewLevel() == PrivilegeLevel.SUPER_ADMIN) {
+      throw new AuthException("User does not have the required privilege level");
+    }
+
+    // check password and if the privilege level is different
+    if (Passwords.isExpectedPassword(
```

Comment:
I think this should be checking the changePrivilegeLevelRequest.getPassword() against the calling user's password, not the target user's password that's being checked here